internal/services/cognitive/cognitive_account_ai_services_resource.go

```diff
@@ -0,0 +1,704 @@
+				Tags: pointer.To(model.Tags),
+			}
+
+			expandIdentity, err := identity.ExpandSystemAndUserAssignedMapFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+			props.Identity = expandIdentity
+
+			if _, err = client.AccountsUpdate(ctx, id, props); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			stateConf := &pluginsdk.StateChangeConf{
+				Pending:    []string{"Accepted"},
+				Target:     []string{"Succeeded"},
+				Refresh:    aiServicesAccountStateRefreshFunc(ctx, client, id),
+				MinTimeout: 15 * time.Second,
+				Timeout:    metadata.ResourceData.Timeout(pluginsdk.TimeoutUpdate),
+			}
+
+			if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for update of %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (AIServicesAccountResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.AccountsClient
+
+			id, err := cognitiveservicesaccounts.ParseAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			log.Printf("[DEBUG] Retrieving %s..", *id)
+			account, err := client.AccountsGet(ctx, *id)
+			if err != nil || account.Model == nil || account.Model.Location == nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			deletedAccountId := cognitiveservicesaccounts.NewDeletedAccountID(id.SubscriptionId, *account.Model.Location, id.ResourceGroupName, id.AccountName)
+			if err != nil {
+				return err
+			}
+
+			log.Printf("[DEBUG] Deleting %s..", *id)
+			if err := client.AccountsDeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+			if metadata.Client.Features.CognitiveAccountAIServices.PurgeSoftDeleteOnDestroy {
+				log.Printf("[DEBUG] Purging %s..", *id)
+				if err := client.DeletedAccountsPurgeThenPoll(ctx, deletedAccountId); err != nil {
+					return fmt.Errorf("purging %s: %+v", *id, err)
+				}
+			} else {
+				log.Printf("[DEBUG] Skipping Purge of %s", *id)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (AIServicesAccountResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return cognitiveservicesaccounts.ValidateAccountID
+}
+
+func aiServicesAccountStateRefreshFunc(ctx context.Context, client *cognitiveservicesaccounts.CognitiveServicesAccountsClient, id cognitiveservicesaccounts.AccountId) pluginsdk.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.AccountsGet(ctx, id)
+		if err != nil {
+			return nil, "", fmt.Errorf("polling for %s: %+v", id, err)
+		}
+
+		if res.Model != nil && res.Model.Properties != nil && res.Model.Properties.ProvisioningState != nil {
+			return res, string(*res.Model.Properties.ProvisioningState), nil
+		}
+		return nil, "", fmt.Errorf("unable to read provisioning state")
+	}
+}
```

Comment:
this is copied from Cognitive Account and I just remove the unnecessary properties and codes

Reply:
thanks, will remove the refresh part.  There's only one API for cognitive account,  keep everything the same is my first choice.

Reply:
@xuzhang3 with respect, you still haven’t answered my question: what’s the API actually doing here? Can you show the HTTP Request/Response of what’s being sent/coming back over the wire, so that we can understand if there’s another issue here?

Reply:
the SDK  can help handle the request, so I removed the state refresh and replaced by `future.Poller.PollUntilDone`